File: backend/Concord/Concord.Application/Models/SubscriptionResponse.cs
using System.Text.Json.Serialization;

namespace Concord.Application.Models;

public class SubscriptionResponse
{
    [JsonPropertyName("Status")]
    public string Status { get; set; } = "None";

    [JsonPropertyName("CurrentPeriodEnd")]
    public DateTime? CurrentPeriodEnd { get; set; }

    [JsonPropertyName("CancelAtPeriodEnd")]
    public bool CancelAtPeriodEnd { get; set; }
}

File: mobile/concord/lib/api/models/subscription_response.dart
import 'enums.dart';
import 'json_utils.dart';

class SubscriptionResponse {
  const SubscriptionResponse({
    required this.status,
    required this.currentPeriodEnd,
    required this.cancelAtPeriodEnd,
  });

  factory SubscriptionResponse.fromJson(Map<String, dynamic> json) {
    return SubscriptionResponse(
      status: SubscriptionStatus.fromWire(json.field('Status')),
      currentPeriodEnd: parseNullableDateTime(json.field('CurrentPeriodEnd')),
      cancelAtPeriodEnd: json.field('CancelAtPeriodEnd') as bool? ?? false,
    );
  }

  final SubscriptionStatus status;
  final DateTime? currentPeriodEnd;
  final bool cancelAtPeriodEnd;
}
